packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -145,24 +165,86 @@ export class DetachedFieldIndex {
 	public createEntry(nodeId?: Delta.DetachedNodeId, count: number = 1): ForestRootId {
 		const root = this.rootIdAllocator.allocate(count);
 		if (nodeId !== undefined) {
-			for (let i = 0; i < count; i++) {
-				assert(
-					tryGetFromNestedMap(
-						this.detachedNodeToField,
-						nodeId.major,
-						nodeId.minor + i,
-					) === undefined,
-					0x7ce /* Detached node ID already exists in index */,
-				);
-				setInNestedMap(this.detachedNodeToField, nodeId.major, nodeId.minor + i, root + i);
-			}
+			this.setDetachedNodeRange(nodeId, count, root);
 		}
 		return root;
 	}
 
+	private setDetachedNodeRange(
+		nodeId: Delta.DetachedNodeId,
+		count: number,
+		root: ForestRootId,
+	): void {
+		if (!this.detachedNodeRangeMap.has(nodeId.major)) {
+			this.detachedNodeRangeMap.set(nodeId.major, []);
+		}
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const innerRangeMap = this.detachedNodeRangeMap.get(nodeId.major)!;
+		assert(
+			getFirstEntryFromRangeMap(innerRangeMap, nodeId.minor, count) === undefined,
+			"The detached node range already exists in the map",
```

Comment:
```suggestion
			"The detached node range already exists in the index",
```
The callers don't need to know what data structure we're using. "The index" is also more specific.